src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -196,40 +187,28 @@ CoreSMTSolver::initialize( )
         break; // Boolean atoms
     }
 
+    if (config.produceProof() && !proof) {
+        proof = std::unique_ptr<Proof>(new Proof(this->ca));
+    }
+
     init = true;
 }
 
 CoreSMTSolver::~CoreSMTSolver()
 {
-#ifdef PRODUCE_PROOF
-
-    for (int i = 0; i < units.size(); i++)   if(units[i] != CRef_Undef)   ca.free(units[i]);
-    for (int i = 0; i < clauses.size(); i++) if(clauses[i] != CRef_Undef) ca.free(clauses[i]);
-    for (int i = 0; i < pleaves.size(); i++) if(pleaves[i] != CRef_Undef) ca.free(pleaves[i]);
-    for (int i = 0; i < learnts.size(); i++) if(learnts[i] != CRef_Undef) ca.free(learnts[i]);
-    for (int i = 0; i < tleaves.size(); i++) if(tleaves[i] != CRef_Undef) ca.free(tleaves[i]);
-
-#else
-    for (int i = 0; i < learnts.size(); i++) ca.free(learnts[i]);
-    for (int i = 0; i < clauses.size(); i++) ca.free(clauses[i]);
-#endif
-
-    for (int i = 0; i < tmp_reas.size(); i++) ca.free(tmp_reas[i]);
-
 #ifdef STATISTICS
     if ( config.produceStats() != 0 ) printStatistics ( config.getStatsOut( ) );
     // TODO added for convenience
     if ( config.print_stats != 0 ) printStatistics ( cerr );
 
 #endif
 
-#ifdef PRODUCE_PROOF
-    if (PRINT_LRA_ITP_STATS && LRA_Interpolator::stats.anyOpportunity()) {
+#ifdef PRINT_DECOMPOSED_STATS
+    if (LRA_Interpolator::stats.anyOpportunity()) {
```

Comment:
The statistics are in the LRAInterpolator. It was just a question when the call to print should occur. I can try to move this to the interpolation code. It was needed only get statistics for the paper, it is not really a public feature (but Sepideh was also using it for some of her statistics).

Reply:
I moved the printing from CoreSolver to LRASolver since LRASolver is also computing the interpolants. Other than that, I don't think there is a good place for it, so the other option would be to just remove it altogether.